src/lib/hasFieldSelection.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable require-yield */
+
+import {
+  FieldNode,
+  visit,
+  FragmentDefinitionNode,
+  BREAK,
+  GraphQLResolveInfo,
+} from "graphql"
+
+export const hasFieldSelection = (
+  resolveInfo: GraphQLResolveInfo,
+  match: (nodeName: string) => boolean
+): boolean => {
+  if (!resolveInfo.fieldNodes) return false
+  const firstFieldNode = resolveInfo.fieldNodes[0]
+
+  let matched: boolean = false
+
+  const visitor = (
+    fieldNode: FieldNode | FragmentDefinitionNode,
+    fragments?: { [key: string]: FragmentDefinitionNode }
+  ): void => {
+    visit(fieldNode, {
+      Field(node, _key, _parent, path, _ancestors) {
+        if (path.length === 3 && match(node.name.value)) {
+          matched = true
+          return BREAK
+        }
+      },
+      FragmentSpread(node) {
+        const fragmentDef = resolveInfo.fragments[node.name.value]
+        visitor(fragmentDef, fragments)
+      },
+    })
+  }
+  visitor(firstFieldNode)
+  return matched
+}
+
+export const includesFieldsSelection = (
+  resolveInfo: GraphQLResolveInfo,
+  fields: string[]
+): boolean => {
+  return hasFieldSelection(resolveInfo, nodeName => fields.includes(nodeName))
+}
+
+export const includesOtherFieldsSelection = (
```

Comment:
…and?

```suggestion
export const hasNoIntersectionWithSelectionSet = (
```

Reply:
ended up renaming but not to this name, `hasNoIntersectionWithSelectionSet` doesn't reflect what it does, this checks to see if there are fields other than passed in list of fields in the query.I thought of super set but then it's not really superset too since its totally fine to have other fields and non of the passed in ones.